src/DevTools/LikelyCulpritDetector.php

```diff
@@ -54,7 +54,7 @@ public function analyze_backtrace() {
 	 */
 	public function analyze_exception( Exception $exception ) {
 		$trace = $exception->getTrace();
-		array_unshift( $trace, [ 'file' => $exception->getFile() ] );
+		array_unshift( $trace, [ FileReflection::SOURCE_FILE => $exception->getFile() ] );
```

Comment:
I was hesitant about these constants. I think the `source` array (which is reused in many different places) should at one point become a `Source` object instead.

These constants here won't hurt for now, and this is all marked as @internal anyway. But the long-term approach should be to refactor all the `source` array shenanigans into an object, so that these constants won't be needed as we'll have methods instead.

Reply:
What we could do as an intermediate step is to extract an interface `Source` that receives these constants. They would then look like `Source::TYPE`, so a bit less unwieldy.

Later on, we can then change this `Source` interface into a `Source` class to turn the arrays into objects (or even keep it as an interface and have differing implementations if that makes sense).

Reply:
The main benefit of that intermediate step is that we not only move the hard-coded strings into constants and thereby have a "catalogue of options", we also do so in such a way that it can be kept with the move to objects. No matter what `Source` evolves into, the constants can stay within it.